spring-cloud-stream/src/test/java/org/springframework/cloud/stream/binder/BinderFactoryConfigurationTests.java

```diff
@@ -148,6 +148,38 @@ public void loadBinderTypeRegistryWithTwoBinders() throws Exception {
 	}
 
 	@Test
+	public void loadBinderTypeRegistryWithUserDefinedBinder() throws Exception {
+
+		ConfigurableApplicationContext context = createBinderTestContext(
+				new String[] { "binder1"},
+				"spring.cloud.stream.binders.custom.type=binder1",
+				"spring.cloud.stream.binders.custom.environment.binder1.name=foo",
+				"spring.cloud.stream.binders.custom.applicationProvided=false",
```

Comment:
This is intentional set to `false`, as `binder1` will still be elected as the default binder.